src/fsharp/fsi/fsi.fs

```diff
@@ -1273,13 +1275,51 @@ type internal FsiDynamicCompiler
         resolutions,
         { istate with tcState = tcState.NextStateAfterIncrementalFragment(tcEnv); optEnv = optEnv }
 
+
+    member __.EvalDependencyManagerTextFragment (packageManager:DependencyManagerIntegration.IDependencyManagerProvider,m,path: string) =
+        let path = DependencyManagerIntegration.removeDependencyManagerKey packageManager.Key path
+        
+        match tcConfigB.packageManagerLines |> Map.tryFind packageManager.Key with
+        | Some lines -> tcConfigB.packageManagerLines <- Map.add packageManager.Key (lines @ [path,m]) tcConfigB.packageManagerLines
+        | _ -> tcConfigB.packageManagerLines <- Map.add packageManager.Key [path,m] tcConfigB.packageManagerLines
+
+        needsPackageResolution <- true
+         
+    member fsiDynamicCompiler.CommitDependencyManagerText (ctok, istate: FsiDynamicCompilerState, lexResourceManager, errorLogger) = 
+        if not needsPackageResolution then istate else
+        needsPackageResolution <- false
+        
+        let istate = ref istate
+        for kv in tcConfigB.packageManagerLines do
+            let packageManagerKey,packageManagerLines = kv.Key,kv.Value
+            match packageManagerLines with
+            | [] -> ()
+            | (_,m)::_ ->
+                let packageManagerTextLines = packageManagerLines |> List.map fst
+                match DependencyManagerIntegration.tryFindDependencyManagerByKey m packageManagerKey with
+                | None ->
+                    errorR(DependencyManagerIntegration.createPackageManagerUnknownError packageManagerKey m)
+                | Some packageManager ->
+                    match DependencyManagerIntegration.resolve packageManager tcConfigB.implicitIncludeDir "stdin.fsx" "stdin.fsx" m packageManagerTextLines with
+                    | None -> () // error already reported
+                    | Some (loadScript,additionalIncludeFolders) -> 
+                        for folder in additionalIncludeFolders do 
+                            tcConfigB.AddIncludePath(m,folder,"")
+
+                        match loadScript with
+                        | Some loadScript -> istate := fsiDynamicCompiler.EvalSourceFiles (ctok, !istate, m, [loadScript], lexResourceManager, errorLogger)
```

Comment:
Instead of using a ref variable here, this can be restructured as a Seq.fold over `tcConfigB.packageManagerLines`